File: hosts/adobe/dx-indesign-uxp/src/commandPlans.ts
(function registerDxInDesignCommandPlans(global) {
  const { DX_INDESIGN_MESSAGES } = global.dxInDesignMessages;

  const DX_INDESIGN_COMMAND_PLANS = {
    showStatus: {
      messageType: DX_INDESIGN_MESSAGES.showStatus,
      operation: "dx.status",
      transport: "local-service",
      requiresRuntimeProof: true,
      mutatesInDesignDocument: false
    },
    searchAssets: {
      messageType: DX_INDESIGN_MESSAGES.searchAssets,
      operation: "dx.assets.search",
      transport: "local-service",
      requiresRuntimeProof: true,
      mutatesInDesignDocument: false
    },
    showReceipts: {
      messageType: DX_INDESIGN_MESSAGES.showReceipts,
      operation: "receipt.showPath",
      transport: "host-ui",
      requiresRuntimeProof: false,
      requiresLoadedHostProof: true,
      mutatesInDesignDocument: false
    }
  };

  function commandPlanForMessage(messageType) {
    return Object.values(DX_INDESIGN_COMMAND_PLANS).find(
      (plan) => plan.messageType === messageType
    );
  }

  global.dxInDesignCommandPlans = {
    DX_INDESIGN_COMMAND_PLANS,
    commandPlanForMessage
  };
})(globalThis);

File: hosts/zed/dx-zed/src/command_plans.rs
pub struct DxZedCommandPlan {
    pub slash_command: &'static str,
    pub label: &'static str,
    pub operation: &'static str,
    pub transport: &'static str,
    pub requires_runtime_proof: bool,
}

pub const DX_ZED_COMMAND_PLANS: &[DxZedCommandPlan] = &[
    DxZedCommandPlan {
        slash_command: "dx-status",
        label: "DX Status",
        operation: "dx.status",
        transport: "local-service",
        requires_runtime_proof: true,
    },
    DxZedCommandPlan {
        slash_command: "dx-assets",
        label: "DX Assets",
        operation: "dx.assets.search",
        transport: "local-service",
        requires_runtime_proof: true,
    },
    DxZedCommandPlan {
        slash_command: "dx-receipts",
        label: "DX Receipts",
        operation: "receipt.showPath",
        transport: "host-ui",
        requires_runtime_proof: false,
    },
];

pub fn command_plan_for(slash_command: &str) -> Option<&'static DxZedCommandPlan> {
    DX_ZED_COMMAND_PLANS
        .iter()
        .find(|plan| plan.slash_command == slash_command)
}
